OutOfSchool/OutOfSchool.WebApi/Services/CurrentUserService.cs

```diff
@@ -186,6 +189,28 @@ private async Task<bool> ProviderAdminHasRights(string providerAdminId)
         return result;
     }
 
+    private async Task<bool> ProviderDeputyHasRights(Guid providerId)
```

Comment:
probably can be merged with provideradmin has rights like parent/child check is working. I will see if there's a good solution to it.